File: mvp/src/main/java/com/wasisto/githubuserfinder/ui/userdetails/UserDetailsView.java
/*
 * Copyright (c) 2018 Andika Wasisto
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.wasisto.githubuserfinder.ui.userdetails;

import android.support.annotation.StringRes;
import com.wasisto.githubuserfinder.ui.View;

public interface UserDetailsView extends View {

    void showAvatar(String avatarUrl);

    void hideAvatar();

    void showName(String name);

    void hideName();

    void showUsername(String username);

    void hideUsername();

    void showCompany(String company);

    void hideCompany();

    void showLocation(String location);

    void hideLocation();

    void showBlog(String blog);

    void hideBlog();

    void openBrowser(String uri);

    void showToast(@StringRes int resId);

    void closeActivity();
}

File: mvp-kotlin/src/main/java/com/wasisto/githubuserfinder/ui/userdetails/UserDetailsView.kt
/*
 * Copyright (c) 2018 Andika Wasisto
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.wasisto.githubuserfinder.ui.userdetails

import androidx.annotation.StringRes

import com.wasisto.githubuserfinder.ui.View

interface UserDetailsView : View {

    fun showAvatar(avatarUrl: String)

    fun hideAvatar()

    fun showName(name: String)

    fun hideName()

    fun showUsername(username: String)

    fun hideUsername()

    fun showCompany(company: String)

    fun hideCompany()

    fun showLocation(location: String)

    fun hideLocation()

    fun showBlog(blog: String)

    fun hideBlog()

    fun openBrowser(uri: String)

    fun showToast(@StringRes resId: Int)

    fun closeActivity()
}
